### ISSUE TITLE
Undefined context

### ISSUE THREAD

**Issue by monolithed** [nodeca/argparse#89]
``` js
parser.addArgument(['-t', '--type'], {
    defaultValue: 'regular',

    type (value) {
        console.log(this) // undefined
    },

    help: 'Account type can be [regular | pdd | external]'
});
```

Why `this` is undefined?

**Comment by puzrin** [comment]
Hm... why should it be defined?

**Comment by monolithed** [comment]
```js
parser.addArgument(['-t', '--type'], {
	defaultValue: 'regular',

	type (value) {
		if (!/^(regular|pdd|external)$/.test(value)) {
			Logger.error(`"-t/--type": ${this.help}`);
		}

		return value;
	},

	help: 'Account type can be [regular | pdd | external]'
});
```

It seems expected and useful behavior, why not?

**Comment by puzrin** [comment]
Don't know. Nobody asked for 3 years. Is it supported in python?

**Comment by monolithed** [comment]
```
import argparse

def type (self, value):
	print(self.help)

	return value

parser = argparse.ArgumentParser()
parser.add_argument('-t', '--type', help='help', type=type)

print(parser.parse_args())
```

```
➜ python options.py --type 1
usage: help.py [-h] [-t TYPE]
help.py: error: argument -t/--type: invalid type value: '1'
```

Expected. Python and JavaScript objects have different behavior

**Comment by puzrin** [comment]
Do you mean `type` function has different signature in js and python?

**Comment by monolithed** [comment]
Yup

**Comment by puzrin** [comment]
Seems that's a bug :( . One more difference to add to readme.

Then i think now it's really better to bind `this`.

**Comment by rlidwka** [comment]
We've ported everything from scratch based on python 3.9 (released as argparse version 2.0).

Please check if it's still relevant in the current version.

In any case, how would you achieve the same result in python's argparse? It should work the same way here.

**Comment by puzrin** [comment]
Let's close as oudated. Feel free to create new issue if needed.
